wallet-new/src/Cardano/Wallet/API/V1/TH.hs

```diff
@@ -0,0 +1,199 @@
+{-# LANGUAGE LambdaCase      #-}
+{-# LANGUAGE QuasiQuotes     #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TupleSections   #-}
+
+module Cardano.Wallet.API.V1.TH
+       ( conNamesList
+       , deriveWalletErrorJSON
+       ) where
+
+import           Universum
+
+import           Data.Aeson
+import           Data.Aeson.Types (Parser)
+import           Data.Char (isUpper, toLower)
+import qualified Data.List as DL
+import           Language.Haskell.TH
+import qualified Serokell.Aeson.Options as Serokell
+
+--
+-- String manipulation utils
+--
+
+headToLower :: String -> String
+headToLower []     = []
+headToLower (x:xs) = toLower x : xs
+
+stripFieldPrefix :: String -> String
+stripFieldPrefix = dropWhile (not . isUpper)
+
+mkJsonKey :: String -> String
+mkJsonKey s =
+    let s' = headToLower $ stripFieldPrefix s
+    in if null s' then s else s'
```

Comment:
Mostly bikeshedding, but if you convert `headToLower` to return `Maybe String` (where `Nothing` is yielded on empty list) you can rewrite `mkJsonKey` as:

```
mkJsonKey s = fromMaybe s (headToLower . stripFieldPrefix $ s)
```